src/vm/class.cpp

```diff
@@ -1282,9 +1281,17 @@ EEClass::CheckForHFA()
             break;
 
         case ELEMENT_TYPE_R4:
+            if (pFD->GetOffset() % alignof(float) != 0) // HFAs don't have unaligned fields.
```

Comment:
We do not want to depend on the C/C++ idea of alignment. It breaks with cross-compilation among other things. The alignment should be actual number.